stdlib/testing/assert_empty.go

```diff
@@ -86,13 +86,30 @@ func (t *AssertEmptyTransformation) RetractTable(id execute.DatasetID, key flux.
 }
 
 func (t *AssertEmptyTransformation) Process(id execute.DatasetID, tbl flux.Table) error {
-	if !tbl.Empty() {
-		t.failures++
+	if tbl.Empty() {
+		// Table is empty so we succeed.
+		// TODO: The Do method must be called at the moment.
+		return tbl.Do(func(cr flux.ColReader) error {
+			return nil
+		})
 	}
-	// TODO: The Do method must be called at the moment.
-	return tbl.Do(func(cr flux.ColReader) error {
+
+	// Confirm that the table is, in fact, not empty.
+	// Some implementations don't know until Do is called.
```

Comment:
Isn't this a problem?  Seems really unintuitive.  Which implementations don't know they are empty until `Do` is called?

Reply:
I think the intention was to fix it at some point, but the influxdb one doesn't for simplicity.

Reply:
Is there an issue for it?  If not, can you file one?

Reply:
I wasn't going to. I didn't really see it as an issue. I thought of the `Empty()` as more of a hint. In general, requiring that a table know it is empty before `Do` is called means that whatever resource it is currently utilizing must be used before `Do` is called.

Although if we choose to view it that way, it should probably be noted in the interface docs.

Reply:
I think that it's fine if `Empty()` returns false in situations where the table is empty, but I created influxdata/influxdb#14193 anyway so we can investigate it and see if it's easy enough to implement.

I changed the godoc comment on empty so that it is true for what is currently true at the moment.

Reply:
The docs state:

```
Empty returns whether the table contains no records.
```

It is not a hint and a table implementation that returns `false` where there are no rows in the table would be an incorrect implementation of the interface.

Yes it requires that the implementation know before Do is called whether there will be at least one record produced. The CSV decoders/encoders implement this correctly and take advantage of it to encode empty tables appropriately.

Reply:
FWIW I believe the InfluxDB source does implement this correctly simply because the storage RPC calls never produce an empty table.